src/app/shared/dovetail/dovetail-segment.ts

```diff
@@ -0,0 +1,96 @@
+import * as urlTemplate from 'url-template';
+import { AdzerkDecision } from '../adzerk';
+import { DovetailResponseEntry } from './dovetail-fetcher';
+
+export class DovetailSegment {
+  
+  id: string;
+  type: string;
+  duration: number;
+  audioUrl: string;
+  unplayable = false;
+
+  private beforeTrackers: string[] = [];
+  private afterTrackers: string[] = [];
+  private pixels: HTMLImageElement[] = [];
+
+  constructor(
+    entry: DovetailResponseEntry, 
+    decision?: AdzerkDecision, 
+    tracker?: any, 
+    countDownload?: boolean
+  ) {
+    this.id = entry.id;
+    this.type = entry.type || 'ad';
+    if (entry.duration && typeof(entry.duration) === 'string') {
+      this.duration = parseFloat('' + entry.duration);
+    } else if (entry.duration && typeof(entry.duration) === 'number') {
+      this.duration = entry.duration;
+    } else {
+      this.duration = 0;
+    }
+    this.audioUrl = entry.audioUrl;
+
+    // optional adzerk data
+    if (decision) {
+      this.audioUrl = decision.contents[0].data.imageUrl;
+      this.duration = 10;
+      // TODO: why only ads?
+      if (this.type === 'ad' || this.type === 'houseAd') {
+        this.afterTrackers.push(decision.impressionUrl);
+      }
+      // TODO: also concat decision.contents[0].data.customData.pingbacks[]
```

Comment:
The scheme we invented for 3rd-party Adzerk pingbacks (implemented in `analytics-ingest-lambda`) is a bit complex.  So I've left this as a TODO for now, hoping at some point to use a shared lib between these 2 projects.